Backend/src/main/java/com/harmonia/backend/controller/ServerController.java

```diff
@@ -7,60 +7,114 @@
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+/**
+ * The ServerController class handles HTTP requests related to server management.
+ * This class maps incoming requests to the corresponding ServerService methods.
+ */
 @RestController
 @RequestMapping("server")
 public class ServerController {
 
+    /**
+     * The ServerService used to perform server-related operations.
+     */
     @Autowired
     private ServerService serverService;
 
-    public ServerController(ServerService serverService){
+    /**
+     * Constructs a new ServerController with the given ServerService.
+     *
+     * @param serverService the ServerService to use
+     */
+    public ServerController(ServerService serverService) {
         this.serverService = serverService;
     }
 
 
+    /**
+     * Returns a list of servers filtered by serverName.
+     *
+     * @param serverName the name of the server to filter by
+     * @return a list of servers with matching serverName
+     */
     @GetMapping("serverName/{serverName}")
     @CrossOrigin
-    public Iterable<Server> listServersByServerName(@PathVariable("serverName") String serverName){
+    public Iterable<Server> listServersByServerName(@PathVariable("serverName") String serverName) {
         return serverService.listServersByServerName(serverName);
     }
 
+
+    /**
+     * Returns a list of servers filtered by serverCategory.
+     *
+     * @param serverCategory the category of the server to filter by
+     * @return a list of servers with matching serverCategory
+     */
     @GetMapping("category/{serverCategory}")
```

Comment:
maybe use  @GetMapping("serverCategory/{serverCategory}") to make it more standard